src/core/Navigation/SideNavigation/SideNavigationItem/SideNavigationItem.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { ReactElement, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlLi, HtmlSpan, HtmlUl } from '../../../../reset';
+import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../../theme';
+import { baseStyles } from './SideNavigationItem.baseStyles';
+import styled from 'styled-components';
+import { Icon } from '../../../Icon/Icon';
+
+export interface SideNavigationItemProps {
+  /** Custom class */
+  className?: string;
+  /** Use the polymorphic RouterLink component as content to get intended CSS styling */
+  content: ReactNode;
+  /** Nested SideNavigationItems */
+  children?: ReactNode;
+  /** Sub-level of the item. 1-3 */
+  subLevel: 1 | 2 | 3;
+  /** Show item as the selected one */
+  selected?: boolean;
+  /** Disable the item */
+  disabled?: boolean;
+  /** Force the item to show its children even when not selected */
+  expanded?: boolean;
+}
+
+const baseClassName = 'fi-side-navigation-item';
+const selectedClassName = `${baseClassName}--selected`;
+const childSelectedClassName = `${baseClassName}--child-selected`;
+const disabledClassName = `${baseClassName}--disabled`;
+const subListClassName = `${baseClassName}_sub-list`;
+const contentWrapperClassName = `${baseClassName}_content-wrapper`;
+
+const BaseSideNavigationItem = ({
+  subLevel,
+  selected,
+  className,
+  children,
+  content,
+  disabled,
+  expanded,
+  ...passProps
+}: SideNavigationItemProps) => {
+  // Loop through all children to check if a child of this item is selected
+  let childIsSelected = false;
+  React.Children.forEach(children, (element) => {
+    if (!React.isValidElement(element)) return;
+
+    if (element.props) {
+      if (element.props.selected) {
+        childIsSelected = true;
+      } else if (element.props.children) {
+        element.props.children.forEach((c: ReactElement) => {
+          if (c.props) {
+            if (c.props.selected) {
+              childIsSelected = true;
+              return;
+            }
+          }
+        });
+      }
+    }
+  });
```

Comment:
I feel like this could be more readable done otherwise. Maybe somewhere along the lines of:

  React.Children.forEach(children, (element) => {
    if (!React.isValidElement(element) || !element.props) return;

    if (element.props.selected) {
      childIsSelected = true;
      return;
    }

    if (!element.props.children) return;

    element.props.children.forEach((child: ReactElement) => {
      if (child.props?.selected) {
        childIsSelected = true;
        return;
      }
    });
  });

Reply:
Fair enough, fewer if-statements and they are not nested. 

Changed it like you suggested!